service/history/shard/controller_impl.go

```diff
@@ -353,9 +353,17 @@ func (c *ControllerImpl) acquireShards() {
 			c.contextTaggedLogger.Error("Unable to create history shard context", tag.Error(err), tag.OperationFailed, tag.ShardID(shardID))
 			return
 		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), shardIOTimeout)
+		defer cancel()
+		if err := shard.AssertOwnership(ctx); IsShardOwnershipLostError(err) {
+			c.CloseShardByID(shardID)
```

Comment:
should it handle non shard ownership lost error?

Reply:
i believe this function should only try to assert shard ownership, if there is a timeout error, other part of the system should handle, or this logic can handle in later iteration